### PR TITLE
perform comments Crud Operations

### PR DESCRIPTION
**Pull request by yogesh-pawar-2023**
- we can post comment
- get comment details
- get list of all comment of appropriate facilitator
- update comment based on comment is
- delete comment based on id

 

LINK :[Code Infra for commenting system](https://tracker.tekdi.net/issues/201639)

### CODE REVIEW

File: src/src/comments/comments.module.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { CommentsService } from './comments.service';
+
+@Module({
+  imports:[HasuraModule,HasuraModuleFromServices,UserModule],
+  controllers: [CommentsController],
+  providers: [CommentsService]
+})
+export class CommentsModule implements NestModule {
+  configure(consumer: MiddlewareConsumer) {
+		consumer.apply(AuthMiddleware).forRoutes('*');
```
Comment (parthgandhitekdi) [inline src/src/comments/comments.module.ts:16]: Apply middleware for `CommentsController` only

File: src/src/comments/comments.service.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+				message: 'Unable to Send Comment!',
+				data: {},
+			});
+		}
+	}
+
+	async findAll(request, context, context_id, resp) {
+		const data = {
+			query: `query MyQuery {
+        comments(where: {_and: {context: {_eq: "${context}"}, context_id: {_eq: ${context_id}}}}){
```
Comment (parthgandhitekdi) [inline src/src/comments/comments.service.ts:58]: Format document properly

File: src/src/comments/comments.service.ts [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+				message: 'Comments found success!',
+				data: result,
+			});
+		}
+	}
+
+	async findOne(id: number, context_id: any, context: any) {
+		const data = {
+			query: `query MyQuery {
+        comments(where: {_and: {id: {_eq: ${id}}, context_id: {_eq:${context_id}}}}){
```
Comment (parthgandhitekdi) [inline src/src/comments/comments.service.ts:98]: Format document properly

File: src/src/comments/comments.module.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { MiddlewareConsumer, Module, NestModule } from '@nestjs/common';
+import { UserModule } from 'src/user/user.module';
```
Comment (parthgandhitekdi) [inline src/src/comments/comments.module.ts:2]: Please remove unused module

File: src/src/comments/comments.service.ts [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+		context: any,
+		context_id: number,
+		body: any,
+		request: any,
+		response: any,
+	) {
+		const user_Id = request.mw_userid;
+		let updatedComment = {};
+		const tableName = 'comments';
+		const result = await this.findOne(id, context_id, context);
```
Comment (parthgandhitekdi) [inline src/src/comments/comments.service.ts:132]: 1. Remove `context` and `context_id` from function args

File: src/src/comments/comments.service.ts [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+				return response.status(400).json({
+					success: false,
+					message: `Comment Not Updated!`,
+					data: {},
+				});
+			}
+		} else {
+			return response.status(400).json({
+				success: false,
+				message: `You Are Not Right Person To Update Comment`,
```
Comment (parthgandhitekdi) [inline src/src/comments/comments.service.ts:161]: Send '**comment not found**' message

File: src/src/comments/comments.controller.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+			});
+		}
+	}
+
+	@Patch('/:context/:context_id/:id')
+	@UseGuards(new AuthGuard())
+	update(
+		@Param('id') id: number,
+		@Param('context') context: string,
+		@Param('context_id') context_id: number,
```
Comment (parthgandhitekdi) [inline src/src/comments/comments.controller.ts:77]: Remove `context` and `context_id` from function params and args

Comment (parthgandhitekdi) [inline src/src/comments/comments.controller.ts:77]: Remove `context` and `context_id` from function params and args

File: src/src/comments/comments.service.ts [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+				return response.status(400).json({
+					success: false,
+					message: `Comment Not Delete!`,
+					data: {},
+				});
+			}
+		} else {
+			return response.status(400).json({
+				success: false,
+				message: `You Are Not Right Person To Delete Comment`,
```
Comment (parthgandhitekdi) [inline src/src/comments/comments.service.ts:202]: Send '**comment not found**' message

File: src/src/comments/comments.service.ts [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+		id: number,
+		context: any,
+		context_id: number,
+		request: any,
+		response: any,
+	) {
+		const user_Id = request.mw_userid;
+		let deleteComment = {};
+		const tableName = 'comments';
+		const commentData = await this.findOne(id, context_id, context);
```
Comment (parthgandhitekdi) [inline src/src/comments/comments.service.ts:177]: Remove `context` and `context_id` from function args

File: src/src/comments/comments.module.ts (hunk 1 above)
Comment (yogesh-pawar-2023) [reply src/src/comments/comments.module.ts:16]: updated

File: src/src/comments/comments.controller.ts (hunk 7 above)
Comment (yogesh-pawar-2023) [reply src/src/comments/comments.controller.ts:77]: updated as per comment

File: src/src/comments/comments.service.ts (hunk 5 above)
Comment (yogesh-pawar-2023) [reply src/src/comments/comments.service.ts:132]: resolved

File: src/src/comments/comments.service.ts (hunk 9 above)
Comment (yogesh-pawar-2023) [reply src/src/comments/comments.service.ts:177]: resolved

File: src/src/comments/comments.service.ts (hunk 8 above)
Comment (yogesh-pawar-2023) [reply src/src/comments/comments.service.ts:202]: resolved

File: src/src/comments/dto/create-comment.dto.ts [hunk 10]
```diff
@@ -0,0 +1 @@
+export class CreateCommentDto {}
```
Comment (manojLondhe) [inline src/src/comments/dto/create-comment.dto.ts:1]: pls update and use dto

File: src/src/comments/dto/update-comment.dto.ts [hunk 11]
```diff
@@ -0,0 +1,4 @@
+import { PartialType } from '@nestjs/mapped-types';
+import { CreateCommentDto } from './create-comment.dto';
+
+export class UpdateCommentDto extends PartialType(CreateCommentDto) {}
```
Comment (manojLondhe) [inline src/src/comments/dto/update-comment.dto.ts:4]: pls update and use dto

File: src/src/comments/comments.service.ts [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+				message: 'Unable to Send Comment!',
+				data: {},
+			});
+		}
+	}
+
+	async findAll(request, context, context_id, resp) {
+		const data = {
+			query: `query MyQuery {
+				        comments(where: {_and:[{context:{_eq:"${context}"}},{context_id:{_eq:${context_id}}}]},order_by:{created_datetime:desc}){
```
Comment (manojLondhe) [inline src/src/comments/comments.service.ts:58]: fix indent, use tab with 4 spaces for one indent level

File: src/src/comments/comments.service.ts [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+                  user_id
+                  comment
+                  context
+                  context_id
+                  created_by
+                  created_datetime
+                  updated_by
+                  updated_datetime
+                  status
+									user_details{
```
Comment (manojLondhe) [inline src/src/comments/comments.service.ts:109]: fix indent, use tab with 4 spaces for one indent level

File: src/src/comments/comments.service.ts [hunk 14]
```diff
@@ -0,0 +1,203 @@
...
+				message: 'Comments found success!',
+				data: result,
+			});
+		}
+	}
+
+	async findOne(id: number) {
+		const data = {
+			query: `query MyQuery {
+                comments(where: {id: {_eq: ${id}}}){
```
Comment (manojLondhe) [inline src/src/comments/comments.service.ts:98]: fix indent, use tab with 4 spaces for one indent level

**Review by manojLondhe — CHANGES_REQUESTED** [review]
fix comments

File: src/src/comments/comments.service.ts (hunk 13 above)
Comment (yogesh-pawar-2023) [reply src/src/comments/comments.service.ts:109]: added

File: src/src/comments/comments.service.ts (hunk 14 above)
Comment (yogesh-pawar-2023) [reply src/src/comments/comments.service.ts:98]: added

**Pull request merged**